### PR TITLE
Abductor console exploit fix

### PR DESCRIPTION
**Pull request by ArcaneDefence**
## About The Pull Request

Adds variables to actions used by the abductor console and adds a timer to be checked in order to prevent exploits

## Why It's Good For The Game

image
This is bad. It could be nothing occurring, or an alien about to rapidly teleport between each of them.
Only god and the abductor on the console knows the difference.

## Changelog
:cl:
fix: The abductor ship telepad requires a moment to cool down between uses, and can no longer teleport the console user to multiple locations.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/abductor/machinery/camera.dm [hunk 1]
```diff
@@ -59,16 +59,24 @@
...
 /datum/action/innate/teleport_in/Activate()
 	if(!target || !iscarbon(owner))
 		return
+	if(on_cooldown)
+		to_chat(owner, "<span class='warning'>You must wait [timeleft(on_cooldown)*0.1] seconds to use the [target] again!</span>")
+		return
 	var/mob/living/carbon/human/C = owner
 	var/mob/camera/aiEye/remote/remote_eye = C.remote_control
 	var/obj/machinery/abductor/pad/P = target
+	on_cooldown = addtimer(VARSET_CALLBACK(src, on_cooldown, null), abductor_pad_cooldown , TIMER_STOPPABLE)
```
Comment (ShizCalev) [inline code/modules/antagonists/abductor/machinery/camera.dm]: curious, do timers have a greater performance cost vs just setting `on_cooldown = world_time + abductor_pad_cooldown` and then comparing world.time >= on_cooldown?

Comment (ArcaneDefence) [reply code/modules/antagonists/abductor/machinery/camera.dm]: I'm too inexperienced and uninformed to answer that.

Comment (spookydonut) [reply code/modules/antagonists/abductor/machinery/camera.dm]: Needs TIMER_UNIQUE as well

Comment (ShizCalev) [reply code/modules/antagonists/abductor/machinery/camera.dm]: `change it to world_time + adbuctor_pad_cooldown` instead please.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/abductor/machinery/camera.dm [hunk 2]
```diff
@@ -59,16 +59,24 @@
...
+///Is used to compare to world.time in order to determine if the action should early return
+	var/on_cooldown
 	name = "Send To"
 	icon_icon = 'icons/mob/actions/actions_minor_antag.dmi'
 	button_icon_state = "beam_down"
 
 /datum/action/innate/teleport_in/Activate()
 	if(!target || !iscarbon(owner))
 		return
+	if(world.time <= on_cooldown)
```
Comment (ShizCalev) [inline code/modules/antagonists/abductor/machinery/camera.dm]: actually my b, this should just b `<`

File: code/modules/antagonists/abductor/machinery/camera.dm [hunk 3]
```diff
@@ -59,16 +59,24 @@
...
+	var/on_cooldown
 	name = "Send To"
 	icon_icon = 'icons/mob/actions/actions_minor_antag.dmi'
 	button_icon_state = "beam_down"
 
 /datum/action/innate/teleport_in/Activate()
 	if(!target || !iscarbon(owner))
 		return
+	if(world.time < on_cooldown)
+		to_chat(owner, "<span class='warning'>You must wait for the [target] to cool down before using it again!</span>")
```
Comment (ShizCalev) [inline code/modules/antagonists/abductor/machinery/camera.dm]: if you want to have it display the seconds for the cooldown again, you can do this btw
`to_chat(owner, "<span class='warning'>You must wait [DisplayTimeText(on_cooldown - world.time)] to use the [target] again!</span>")`

File: code/modules/antagonists/abductor/machinery/camera.dm [hunk 4]
```diff
@@ -59,16 +59,24 @@
 	return HAS_TRAIT(H, TRAIT_ABDUCTOR_SCIENTIST_TRAINING)
 
 /datum/action/innate/teleport_in
+///Is the amount of time required between uses
+	var/abductor_pad_cooldown = 8 SECONDS
+///Is used to compare to world.time in order to determine if the action should early return
+	var/on_cooldown
```
Comment (Rohesie) [inline code/modules/antagonists/abductor/machinery/camera.dm]: `on_cooldown` sounds like a bool. Could you try something like `cooldown_time`, `next_use_time` or the likes?

**Pull request merged**